src/validation.h

```diff
@@ -189,32 +189,36 @@ void PruneBlockFilesManual(CChainState& active_chainstate, int nManualPruneHeigh
 * Validation result for a single transaction mempool acceptance.
 */
 struct MempoolAcceptResult {
-    /** Used to indicate the results of mempool validation,
-    * including the possibility of unfinished validation.
+    /** Used to indicate the results of mempool validation.
+    * It's possible for a result to be unknown in the case of
+    * package validation when an earlier tx fails and validation
+    * is terminated early. See ResultType::UNFINISHED.
     */
     enum class ResultType {
         VALID, //!> Fully validated, valid.
         INVALID, //!> Invalid.
+        UNFINISHED, //!> Not fully validated.
     };
-    ResultType m_result_type;
-    TxValidationState m_state;
+    const CTransaction& m_tx;
+    const ResultType m_result_type;
+    const TxValidationState m_state;
 
     // The following fields are only present when m_result_type = ResultType::VALID
     /** Mempool transactions replaced by the tx per BIP 125 rules. */
-    std::optional<std::list<CTransactionRef>> m_replaced_transactions;
-    /** Raw base fees. */
-    std::optional<CAmount> m_base_fees;
+    const std::optional<std::list<CTransactionRef>> m_replaced_transactions;
+    /** Raw base fees in satoshis. */
+    const std::optional<CAmount> m_base_fees;
 
     /** Constructor for failure case */
-    explicit MempoolAcceptResult(TxValidationState state)
-        : m_result_type(ResultType::INVALID),
+    explicit MempoolAcceptResult(const CTransaction& tx, TxValidationState state, bool finished=true)
```

Comment:
I wonder if we should have a separate constructor for the UNFINISHED state. It doesn't really make sense to pass in a `TxValidationState` object if acceptance testing is incomplete.